astropy/coordinates/baseframe.py

```diff
@@ -1625,24 +1621,23 @@ def apply_method(value):
 
                 setattr(new, _attr, value)
 
+        shapes = [
+            getattr(new, "_" + attr).shape
+            for attr in new.frame_attributes
+            if (
+                attr not in new._attr_names_with_defaults
+                and getattr(getattr(new, "_" + attr), "shape", ())
+            )
+        ]
+
         if self.has_data:
             new._data = apply_method(self.data)
+            if getattr(new._data, "shape", ()):
```

Comment:
If this whole part is in fact needed, thenI think we can use that, as mentioned above,  it is impossible for `data` not to have a shape.